wdl-grammar/src/tree.rs

```diff
@@ -690,3 +715,353 @@ impl fmt::Debug for SyntaxTree {
         self.0.fmt(f)
     }
 }
+
+/// Gathers substantial trivia (comments and blank lines) from a
+/// [`SyntaxToken`].
+///
+/// Whitespace is only considered substantial if it contains more than one
+/// newline. Comments are always considered substantial.
+fn gather_substantial_trivia(
```

Comment:
I have some thoughts on improving this function (and the `SyntaxTokenExt` trait), but I think it'd be best to hold off on those for this PR and submit a follow-up.